ci/src/mainnet_revisions/mainnet_revisions.py

```diff
@@ -0,0 +1,189 @@
+    saved_versions = get_saved_versions(repo_root=repo_root)
+    return saved_versions.get("subnets", {}).get(app_subnet_id, "")
+
+def get_subnet_replica_version(subnet_id: str) -> str:
+    ic_admin_path = os.getenv("IC_ADMIN_PATH", "ic-admin")
+    output = subprocess.run(
+        [ic_admin_path, "--nns-urls", "https://ic0.app" ,"get-subnet", subnet_id],
+        text=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+
+    if output.returncode != 0:
+        print("Failed to get subnet", subnet_id, "Error:", output.stderr)
+        exit(1)
+
+    response = json.loads(output.stdout)
+    return response["records"][0]["value"]["replica_version_id"]
+
+def get_repo_root() -> str:
+    return subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"],
+        text=True,
+        stdout=subprocess.PIPE
+    ).stdout.strip()
+
+def main():
+    """Do the main work."""
+
+    class HelpfulParser(argparse.ArgumentParser):
+        """An argparse parser that prints usage on any error."""
+
+        def error(self, message):
+            sys.stderr.write("error: %s\n" % message)
+            self.print_help()
+            sys.exit(2)
+
+    parser = HelpfulParser()
+
+    parser.add_argument("--verbose", "-v", action="store_true", help="Verbose mode")
+
+    args = parser.parse_args()
+
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+
+    current_nns_version = get_subnet_replica_version(nns_subnet_id)
+    logging.info("Current NNS subnet (%s) revision: %s", nns_subnet_id, current_nns_version)
+    current_app_subnet_version = get_subnet_replica_version(app_subnet_id)
+    logging.info("Current App subnet (%s) revision: %s", app_subnet_id, current_app_subnet_version)
+
+    ic_repo_push_token = os.environ.get("PUSH_TO_IC", "ENV_VAR_NOT_SET")
+    repo = "dfinity/ic"
+    remote_url = f"https://oauth2:{ic_repo_push_token}@github.com/{repo}.git"
+
+    repo_root = pathlib.Path(get_repo_root())
+
+    running_on_ci = os.environ.get("CI") or os.environ.get("GITHUB_ACTION")
+    if running_on_ci:
+        logging.info("Setting up git credentials for CI")
+        subprocess.check_call(["git", "config", "--global", "credential.helper", "store --file ~/.git-credentials"])
+        with open(os.path.expanduser("~/.git-credentials"), "w") as f:
+            f.write(f"{remote_url}\nusername=oauth2\npassword={ic_repo_push_token}\n")
+    else:
+        logging.info("Running locally, using default git credentials of the current user")
+
+    if not repo_root.exists():
+        subprocess.check_call(["git", "clone", "--depth=50", remote_url, repo_root])
+    else:
+        subprocess.check_call(["git", "fetch", "origin", "--force"], cwd=repo_root)
+        subprocess.check_call(["git", "checkout", "master"], cwd=repo_root)
+        subprocess.check_call(["git", "reset", "--hard", "origin/master"], cwd=repo_root)
```

Comment:
Why is this necessary if actions/checkout has already checked out the master branch?

Reply:
Because its bazel and everything from it is run in a sandbox environment. I could hack it with `--sandbox_add_mount_pair` but I would rather not.

Reply:
Are you sure? `bazel build/test` will indeed run the build or test in a sandbox but IIUC `bazel run` will just build the binary and then run it in the calling environment.

Reply:
Although according to https://bazel.build/docs/user-manual#running-executables the binary will run in the runfiles tree of the binary. But `BUILD_WORKSPACE_DIRECTORY` will point to the root of the workspace.

Reply:
Actually `BUILD_WORKING_DIRECTORY` would be better to use since it points to the directory where bazel was invoked from which should be the repo you just checked out.

Reply:
My only issue with that is that if somebody is trying to run it from local they basically can't because it will checkout master and remove their work. Adding exceptions to that basically adds unneeded complexity. What do you think?

Reply:
Ok, I understand the concern that you might want to run this locally and don't want the script to modify your current branch.

Then I'll let you decide what to do. I'm fine either way I guess.

Reply:
I decided to have a really easy running experience of just using: 
```bash
python ci/src/mainnet_revisions/mainnet_revisions.py
```

It does introduce a dep to public dashboard but the worst that can happen is that it times out so I guess its fine